src/table/use-table-focus-navigation.ts

```diff
@@ -0,0 +1,166 @@
+
+  const maxColumnIndex = useMemo(() => focusableColumns.length - 1, [focusableColumns]);
+  const minColumnIndex = useMemo(() => (selectionType === 'none' ? 1 : 0), [selectionType]);
+
+  const focusCell = useCallback(
+    (rowIndex: number, columnIndex: number) => {
+      if (tableRoot?.current) {
+        iterateTableCells(tableRoot.current, (cell, rIndex, cIndex) => {
+          if (rIndex === rowIndex && cIndex === columnIndex) {
+            const editButton = cell.querySelector('button:last-child') as HTMLButtonElement | null;
+            editButton?.focus?.({ preventScroll: true });
+          }
+        });
+      }
+    },
+    [tableRoot]
+  );
+
+  const shiftFocus = useCallback(
+    (vertical: -1 | 0 | 1, horizontal: -1 | 0 | 1) => {
+      // istanbul ignore if next
+      if (!currentFocusCell.current) {
+        return;
+      }
+      const [rowIndex, columnIndex] = currentFocusCell.current.slice();
+      let newRowIndex = rowIndex;
+      let newColumnIndex = columnIndex;
+
+      if (vertical !== 0) {
+        newRowIndex = Math.min(numRows, Math.max(rowIndex + vertical, 0));
+      }
+
+      if (horizontal !== 0) {
+        while (newColumnIndex <= maxColumnIndex && newColumnIndex >= minColumnIndex) {
+          newColumnIndex += horizontal;
+          if (focusableColumns[newColumnIndex]) {
+            break;
+          }
+        }
+      }
+
+      if (
+        (rowIndex !== newRowIndex || columnIndex !== newColumnIndex) &&
+        currentFocusCell.current &&
+        tableRoot.current
+      ) {
+        focusCell(newRowIndex, newColumnIndex);
+      }
+    },
+    [focusCell, focusableColumns, maxColumnIndex, minColumnIndex, numRows, tableRoot]
+  );
+
+  const handleArrowKeyEvents = useCallback(
+    (event: KeyboardEvent) => {
+      const abort =
+        !!tableRoot.current?.querySelector('[data-inline-editing-active = "true"]') ||
+        !document.activeElement?.closest('[data-inline-editing-active]');
+
+      if (abort) {
+        return;
+      }
+      switch (event.key) {
+        case 'ArrowUp':
+          event.preventDefault();
+          shiftFocus(-1, 0);
+          break;
+        case 'ArrowDown':
+          event.preventDefault();
+          shiftFocus(1, 0);
+          break;
+        case 'ArrowLeft':
+          event.preventDefault();
+          shiftFocus(0, -1);
+          break;
+        case 'ArrowRight':
+          event.preventDefault();
+          shiftFocus(0, 1);
+          break;
+        // istanbul ignore next (default case = do nothing, not testable)
+        default:
+          return;
+      }
+    },
+    [shiftFocus, tableRoot]
+  );
+
+  useEffect(() => {
+    const eventListeners = new Map<[number, number], { focusin(evt: any): void }>();
+    // istanbul ignore if
+    if (!tableRoot.current) {
+      return;
+    }
+
+    const tableElement = tableRoot.current;
+
+    // istanbul ignore next (tested in use-focus-navigation.test.tsx#L210)
```

Comment:
If it is tested, then ignore is not needed, because there is coverage. If not tested, then this comment is misleading

Reply:
Its tested in integration, but coverage doesn't seem to pick that up

Reply:
[The linked file](https://github.com/fa7ad/components/blob/feat/table-cell-editing/src/table/__tests__/use-focus-navigation.test.tsx) is a unit test, should work fine. Did you mean to link a different file?

Reply:
Yeah I think I wrote the wrong file name